### ISSUE TITLE
Bug in authentication

### ISSUE THREAD

**Issue by MaddalenaTech** [cosullivan/SmtpServer#31]
The default .net SmtpClient produce an AUTH command recognized as invalid by SmtpServer.
The problem is in the base64 password which is not recognized as base64

Ex:
AUTH login bWF0dGVvQHBoYXNjb2RlLm9yZw==

**Comment by cosullivan** [comment]
Hi, 

There are a few different issues here;

a) There is a bug in the parsing of the Base64. I've created a failing test and will fix this.
b) There was also a bug in parsing the the single line authentication method and I have just fixed that.
c) Are you sure that the client is sending "login" here and not "plain"? According to the RFC's, I don't think it is allowed to send a single line with the "login" method, this is only allowed with the "plain" method.

Thanks,
Cain.
